content/app/ui/popups/goto-file-popup.js

```diff
@@ -0,0 +1,104 @@
+define("content/app/ui/popups/goto-file-popup", function(require, exports, module) {
+
+	// imports
+	var constants = require("shared/constants");
+	var KeyCodes = constants.KeyCodes;
+	var DomEvents = constants.DomEvents;
+	var Event = require("content/app/events/event");
+	var EventDispatcher = require("content/app/events/event-dispatcher");
+	var AbstractPopup = require("content/app/ui/popups/abstract-popup");
+
+	/**
+	 * Popup for getting file name from user.
+	 */
+	var GotoFilePopup = Class(AbstractPopup, {
+
+		/**
+		 * GotoFilePopup constructor.
+		 */
+		constructor : function () {
+			GotoFilePopup.$super.call(this);
+
+			this._input = null;
+			this._onBlurHandler = this._onBlur.bind(this);
+			this._onKeyUpHandler = this._onKeyUp.bind(this);
+
+			this._init();
+		},
+
+		/**
+		 * Init popup.
+		 */
+		_init : function () {
+			// create input
+			this._input = document.createElement('input');
+			this._input.type = 'text';
+			this._input.className = 'popup-goto-file-input';
+			this.view.appendChild(this._input);
+			// create label
+			var label = document.createElement('div');
+			label.className = 'popup-goto-file-label';
+			this.view.appendChild(label);
```

Comment:
:+1: 

I've added it, however it is a pure copy of your css. As I mentioned, we should introduce LESS, it will simplify things.